client/src/components/LinkPicker/LinkPicker.js

```diff
@@ -1,31 +1,63 @@
 /* eslint-disable */
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import LinkPickerMenu from './LinkPickerMenu';
-import LinkPickerTitle from './LinkPickerTitle';
-
-const LinkPicker = ({ title, description, typeTitle, types, onSelect, onEdit, onClear }) => (
-  <div className={classnames('link-picker', 'form-control', {'link-picker--selected': typeTitle ? true : false})}>
-    {!typeTitle && <LinkPickerMenu types={types} onSelect={onSelect} /> }
-    {typeTitle && <LinkPickerTitle
-      title={title}
-      description={description}
-      typeTitle={typeTitle}
-      onClear={onClear}
-      onClick={() => onEdit()}
-    />}
-  </div>
-);
+import LinkType from 'types/LinkType';
+import LinkModalContainer from 'containers/LinkModalContainer';
+
+/**
+ * Component which allows users to choose a type of link to create, and opens a modal form for it.
+ */
+const LinkPicker = ({ types, onModalSuccess, onModalClosed }) => {
+  const [typeKey, setTypeKey] = useState('');
+
+  /**
+   * When a link type is selected, set the type key so we can open the modal.
+   */
+  const doSelect = (key) => {
+    setTypeKey(key);
+  }
+
+  /**
+   * Callback for when the modal is closed by the user
+   */
+  const onClosed = () => {
+    if (typeof onModalClosed === 'function') {
+      onModalClosed();
+    }
+    setTypeKey('');
+  }
+
+  /**
+   * Callback for when the modal successfully saves a link
+   */
+  const onSuccess = (value) => {
+    setTypeKey('');
+    onModalSuccess(value);
+  }
+
+  const shouldOpenModal = Boolean(typeKey);
```

Comment:
```suggestion
  const shouldOpenModal = typeKey !== '';
```
Save having to go up and check cases when this would be true/false

Reply:
I don't understand the reasoning but it doesn't make any material difference so I've changed it to avoid ping pong.